lib/command.js

```diff
@@ -185,7 +184,15 @@ module.exports = function command (yargs, usage, validation, globalMiddleware) {
     if (typeof commandHandler.builder === 'function') {
       // a function can be provided, which builds
       // up a yargs chain and possibly returns it.
-      innerYargs = commandHandler.builder(yargs.reset(parsed.aliases))
+      if (commandHandler.builder.length < 2) {
+        innerYargs = await commandHandler.builder(yargs.reset(parsed.aliases))
+      } else {
+        // builder function using a done callback
+        innerYargs = await new Promise(resolve =>
+          commandHandler.builder(yargs.reset(parsed.aliases), resolve)
```

Comment:
What about error? I think it should be something like this:
```
 await new Promise(
     (resolve, reject) =>
          commandHandler.builder(yargs.reset(parsed.aliases),(err, result) => { 
             if (err) return reject(err); 
             resolve(result);
         })
)
```

But better to use `promisify` ofc

Reply:
What is the type of `commandHandler.builder` in this codepath?  What is the type of `commandHandler`?

Reply:
You're right about error handling:
- we should use the standard JS callback format
- if doing so, we should use promisify instead of reinventing the wheel

I'll add some commit to change that (and not only for builder functions).

commandHandler.builder is a function in this codepath (checked a few lines above).

Reply:
done